src/main/java/net/neoforged/neoforge/server/command/DataComponentCommand.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright (c) NeoForged and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.neoforge.server.command;
+
+import com.mojang.brigadier.Command;
+import com.mojang.brigadier.builder.ArgumentBuilder;
+import com.mojang.brigadier.context.CommandContext;
+import com.mojang.brigadier.exceptions.CommandSyntaxException;
+import java.util.Objects;
+import java.util.Optional;
+import net.minecraft.ChatFormatting;
+import net.minecraft.commands.CommandSourceStack;
+import net.minecraft.commands.Commands;
+import net.minecraft.core.component.DataComponentMap;
+import net.minecraft.core.component.DataComponentPatch;
+import net.minecraft.core.component.DataComponentType;
+import net.minecraft.core.registries.BuiltInRegistries;
+import net.minecraft.network.chat.Component;
+import net.minecraft.network.chat.HoverEvent;
+import net.minecraft.network.chat.MutableComponent;
+import net.minecraft.server.level.ServerPlayer;
+import net.minecraft.world.item.ItemStack;
+
+class DataComponentCommand {
+    public static ArgumentBuilder<CommandSourceStack, ?> register() {
+        return Commands.literal("data_components")
+                .then(Commands.literal("list")
+                        .requires(cs -> cs.hasPermission(Commands.LEVEL_GAMEMASTERS))
+                        .executes(DataComponentCommand::listComponents));
+    }
+
+    @SuppressWarnings("OptionalAssignedToNull")
+    private static int listComponents(CommandContext<CommandSourceStack> ctx) throws CommandSyntaxException {
+        ServerPlayer player = ctx.getSource().getPlayerOrException();
+        ItemStack stack = player.getMainHandItem();
+        if (stack.isEmpty()) {
+            ctx.getSource().sendFailure(Component.translatable("commands.neoforge.data_components.list.error.held_stack_empty"));
+            return 0;
+        }
+
+        ctx.getSource().sendSuccess(() -> {
+            MutableComponent text = Component.translatable("commands.neoforge.data_components.list.title", stack.getItem().getName(stack));
```

Comment:
Why `stack.getItem().getName(stack)` instead of `stack.getDisplayName()`?

Reply:
The idea was to show the item's unmodified name (i.e. without influence of a written book's title or the `CUSTOM_NAME` data component), at least to whatever degree that is possible considering `Item#getName()` can be overriden.

Reply:
Alright then. Could you add that as a comment to that line?